linera-sdk/src/lib.rs

```diff
@@ -75,9 +75,11 @@ pub struct ViewStateStorage<A>(std::marker::PhantomData<A>);
 #[async_trait]
 pub trait Contract: Sized {
     /// Message reports for application execution errors.
-    type Error: Error;
+    type Error: Error + From<bcs::Error>;
     /// The desired storage backend to use to store the application's state.
     type Storage;
+    /// Initialization Arguments.
+    type InitializationArguments: FromBcsBytes + Send;
```

Comment:
The convenient trait `FromBcsBytes` becomes unnecessary because users won't see it. (In other words, we might just use `serde::Deserialize` here.)